### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by jared-jewitt** [mswjs/data#229] [General]
Hey @kettanaito, first off I just want to say this library - especially in conjunction with MSW - is amazing! 

I've noticed in a few of your responses here you've mentioned you're in the process of a full rewrite. I'm curious if you have a written roadmap where we can see what's to be expected with the rewrite? I've been using MSW data for a startup of mine and have ran into a few limitations that I'm curious will get implemented or not - particularly re-seeding after deleting records.

**Comment by kettanaito** [comment]
Hey, @jared-jewitt. Thank you for your kinds words.

The rewrite I've mentioned is largely internal, I'm not planning on introducing new features as a part of it. Well, maybe static data generation would be one as I'm approaching the new architecture from a static data perspective and making the queryable functionality an extension of the static functionality. 

That being said, I'm trying to design the library so that it'd allow for many other features that are currently rather tedious to get right (like nullable nested objects, for example). 

I usually open a pull request and include a "Roadmap" section in it directly. I haven't got to that part with the mentioned rewrite yet. I'm looking into some other matters right now and will come back to Data in the future.

**Comment by kettanaito** [comment]
## Update

The v1.0 rewrite has been complete! Please see the updated `README.md` and the release notes for more information. This is a breaking change, so take a look at `MIGRATING.md` for the guidelines on how to migrate from v0.
